lua/entities/magicbutton.lua

```diff
@@ -295,75 +307,350 @@ function ButtonMoneyPrize()
+    ["Near the cabin"] = Vector(-107, 2930, 97),
+    ["Near SportZone"] = Vector(1667, -1918, 85),
+}
+
+local function MagicOutcomeButtonSpawn(ply)
+    local button = ents.Create("magicbutton")
+    button:Spawn()
+    local trace = button:FindHidingSpot()
+
+    if (trace) then
+        button:MoveToTraceResult(trace)
+    end
+
+    if (IsValid(button)) then
+        local loc = Location.Find(button)
+        local locd = Location.GetLocationByIndex(loc or -1)
+        local locname = locd.Name
+
+        if (locname == "Outside") then
+            local nearest = 1000000
+            local nearestname
+            local secondnearestname
+
+            for k, v in pairs(landmarks) do
+                if (v:Distance(button:GetPos()) < nearest) then
+                    secondnearestname = nearestname
+                    nearestname = k
+                    nearest = v:Distance(button:GetPos())
+                end
+            end
+
+            locname = locname .. " (" .. nearestname .. ")"
+        end
 
-    return ""
+        return "and spawned [rainbow2];weewoo;another button;weewoo;[fbc], which appeared somewhere in the location:[white]" .. (locname or "Somewhere stupid")
+    end
+end
+
+local function MagicOutcomeSpawnObject(ply, button)
+    local classes = {
+        sent_ball = {
+            "Bouncy Ball", 10, function(ent)
+                ent:SetBallSize(24)
+                ent:GetPhysicsObject():SetVelocity(VectorRand() * 20)
+            end
+        },
+        npc_headcrab = {
+            "Headcrab", math.random(1, 5), function(ent, button)
+                local pos = table.Random(navmesh.Find(button:GetPos() + button:GetUp() * 50, 100, 2000, 2000)):GetRandomPoint()
+                ent:SetPos(pos)
+            end
+        },
+        npc_grenade_frag = {
+            "Live Grenade", math.random(1, 10), function(ent, button)
+                ent:GetPhysicsObject():SetVelocity(VectorRand() * 100)
+                ent:Fire("SetTimer", math.random(2, 6))
+            end
+        },
+        dodgeball = {
+            "Dodgeball", 1, function(ent, button)
+                ent:GetPhysicsObject():SetVelocity(button:GetUp() * 500)
+            end
+        },
+    }
+
+    local dat, class = table.Random(classes)
+    local number = dat[2] or 1
+    local name = dat[1]
+    local func = dat[3]
+
+    for i = 1, number do
+        local ent = ents.Create(class)
+        ent:Spawn()
+        ent:SetPos(button:GetPos() + button:GetUp() * ent:BoundingRadius())
+        SafeRemoveEntityDelayed(ent, 60 * 5) --whatever we spawn, make sure it's gone after 5 minutes
+
+        if (func) then
+            func(ent, button)
+        end
+    end
+
+    local what = name or ent.PrintName or ent:GetClass()
+
+    if (number > 1) then
+        return "and it spawned " .. number .. " " .. what .. "s!"
+    else
+        return "and it spawned a " .. what .. "!"
+    end
+end
+
+local function MagicOutcomeOverlay(ply, button)
```

Comment:
side comment: i think a lot of these funny random things should be triggered by the mystery box too, since that item gets used a lot more than people will find these buttons anyway. I know i need to expose that code though.